javascript/javascript2/week1/js2w1-hw4.js

```diff
@@ -0,0 +1,65 @@
+// 4 Event types - Optional and a little tricky // 
+console.log('*HW4 4: Event types - Optional and a little tricky*');
+
+let spiritAnimalNames = ['The crying butterfly', ' The old wolf', ' The diminishing tiger', 'The gigantic giraff', 'The old elefant',];
+
+let myButton = document.querySelector('#myButton');
+let inputField = document.querySelector('#textinput');
+
+// to trigger the function if the radio input is selected and then to add input eventlistener to the button 
+
+// the following code is refactored by the help of zendex study room . millions thanks to Ante for the coding journey !!
+function inputFunction() {
+    inputField.addEventListener('input', function () {
+        if (document.querySelector('#radioInputID').checked) {
+            let username = document.querySelector('#textinput').value;
+            if (username === "") {
+                alert('nanme can not be empty input text');
+            } else {
+                let randomSpirit = Math.floor((Math.random() * spiritAnimalNames.length));
+                let sentence = username + '-' + spiritAnimalNames[randomSpirit];
+                document.querySelector('.displayRadioinput1').innerHTML = ' ' + sentence;
+            }
+        }
+    });
+}
+// to trigger the function if the radio click is selected and then to add click eventlistener to the button 
+function clickFuction() {
+    myButton.addEventListener('click', function () {
+        if (document.querySelector('#radioClickID').checked) {
+            let username = document.querySelector('#textinput').value;
+            if (username === "") {
+                alert('name can not be empty click');
+            } else {
+                let randomSpirit = Math.floor((Math.random() * spiritAnimalNames.length));
+                const sentence = username + '-' + spiritAnimalNames[randomSpirit];
+                document.querySelector('.displayClick1').innerHTML = ' ' + sentence;
+            }
+        }
+    });
+}
+// to trigger the function if the radio input is selected and then to add mouse over eventlistener to the button 
+function MouseoverFunction() {
```

Comment:
addEventListener will work when you perform the actions, thats when you click, hover, etc. you don't need to enclose addEventListener inside these functions

Reply:
hi dinythomas89 . thank you for your time and feedback. 
yes i agree with you. i just didi like that because the same button is used for click and mouce over so just to separate to eventlistener event is wrapped into the dediated function. Or may be i didn't get you correctly. can you please eaborate a little bit so it is also clear to me

Reply:
the eventListener will only work when you did that particular action like click on a particular element maybe button.  So when you click on a button the eventListener for click will work and  when you mouse over a button, only  'mouse over' event will take place even if the button are same or different
 Did you try it using functions and without using functions for each events?

Reply:
ok . i checked it and i removed the unnecessary function wrap-up. hope this is ok now . thank you so much !